document/src/main/java/com/ritense/document/repository/DocumentDefinitionRepository.java

```diff
@@ -33,6 +33,8 @@ public interface DocumentDefinitionRepository<T extends DocumentDefinition> exte
 
     Optional<T> findFirstByIdNameOrderByIdVersionDesc(String documentDefinitionName);
 
+    Optional<T> findById(DocumentDefinition.Id documentDefinitionId);
```

Comment:
You don't need this override. It's already part of the `CrudRepository` interface